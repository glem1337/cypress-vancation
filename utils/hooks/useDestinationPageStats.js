import { useSelector, useDispatch } from 'react-redux';

import { MAPBOX_FEATURE_TYPE } from 'constants/searchDestinations';
import {
  searchDestinationParamsSelector,
  searchDestinationFiltersSelector,
  desktopMapVisibilitySelector,
} from 'state/concepts/search-destinations/selectors';
import isPresent from 'utils/isPresent';
import {
  clearAllFilter as clearAllFilterAction,
  toggleDesktopMapVisibility as toggleDesktopMapVisibilityAction,
} from 'state/concepts/search-destinations/actions';
import { loadingSelector } from 'state/data/selectors';
import { fetchCampersEndpoint } from 'state/concepts/campervan-rental/endpoints';
import { campersSelector } from 'state/concepts/campervan-rental/selectors';

function useDestinationPageStats() {
  const dispatch = useDispatch();

  const campers = useSelector(campersSelector);

  const searchParams = useSelector(searchDestinationParamsSelector);

  const persistedFilters = useSelector(searchDestinationFiltersSelector);

  const isDesktopMapVisible = useSelector(desktopMapVisibilitySelector);

  const areCampersFetching = useSelector(
    state => loadingSelector(state, fetchCampersEndpoint.endpoint),
  );

  /**
   * Is any filter applied.
   */
  const isAnyFilterApplied = () => {
    const keys = Object.keys(persistedFilters);

    for (let i = 0; i < keys.length; i += 1) {
      const key = keys[i];

      if (
        key === 'uuid'
        || key === 'isMobileVisible'
      ) {
        // eslint-disable-next-line no-continue
        continue;
      }

      const value = persistedFilters[key];

      if (isPresent(value)) {
        return true;
      }
    }

    return false;
  };

  /**
   * Clear all filters.
   */
  const clearAllFilters = () => {
    dispatch(clearAllFilterAction());
  };

  /**
   * Toggle desktop map visibility
   */
  const toggleDesktopMapVisibility = () => {
    dispatch(toggleDesktopMapVisibilityAction());
  };

  return {
    isSearchResultsPage: searchParams?.location?.type === MAPBOX_FEATURE_TYPE,
    isAnyFilterApplied: isAnyFilterApplied(),
    clearAllFilters,
    isDesktopMapVisible,
    toggleDesktopMapVisibility,
    areCampersFetching,
    campers,
  };
}

export default useDestinationPageStats;
