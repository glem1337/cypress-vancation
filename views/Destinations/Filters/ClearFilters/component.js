import React from 'react';
import { Divider, Button } from 'antd';
import { FormattedMessage } from 'react-intl';

import useDestinationPageStats from 'utils/hooks/useDestinationPageStats';

const ClearFilters = () => {
  const {
    isAnyFilterApplied,
    clearAllFilters,
  } = useDestinationPageStats();

  if (!isAnyFilterApplied) {
    return null;
  }

  return (
    <>
      <Divider type="vertical" />
      <Button
        type="simple-text"
        size="large"
        onClick={clearAllFilters}
      >
        <FormattedMessage id="shared.clearFilters" />
      </Button>
    </>
  );
};

export default ClearFilters;
