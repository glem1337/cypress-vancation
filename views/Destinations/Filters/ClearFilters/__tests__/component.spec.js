import React from 'react';
import { shallow } from 'enzyme';

import useDestinationPageStats from 'utils/hooks/useDestinationPageStats';

import ClearFilters from '../component';

const mockedHookData = {
  isAnyFilterApplied: false,
  clearAllFilters: jest.fn(),
};
jest.mock('utils/hooks/useDestinationPageStats', () => jest.fn(() => mockedHookData));

describe('ClearFilters component tests', () => {
  let component = shallow(<ClearFilters />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isAnyFilterApplied equals true', () => {
    useDestinationPageStats.mockReturnValueOnce({
      isAnyFilterApplied: true,
      clearAllFilters: jest.fn(),
    });

    component = shallow(<ClearFilters />);

    expect(component).toMatchSnapshot();
  });
});
