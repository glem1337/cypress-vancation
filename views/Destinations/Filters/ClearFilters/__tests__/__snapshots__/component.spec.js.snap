// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearFilters component tests matches snapshot 1`] = `""`;

exports[`ClearFilters component tests matches snapshot when isAnyFilterApplied equals true 1`] = `
<Fragment>
  <Divider
    type="vertical"
  />
  <Button
    onClick={[MockFunction]}
    size="large"
    type="simple-text"
  >
    <Memo(FormattedMessage)
      id="shared.clearFilters"
    />
  </Button>
</Fragment>
`;
